src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/PackageCatalogWrapper.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Windows.ApplicationModel;
+using Windows.Foundation;
+
+namespace Microsoft.Plugin.Program.Programs
+{
+
+    /// <summary>
+    /// This is a simple wrapper class around the PackageCatalog to facilitate unit testing.
+    /// </summary>
+    internal class PackageCatalogWrapper : IPackageCatalog
+    {
+        PackageCatalog _packageCatalog;
+
+        public PackageCatalogWrapper()
+        {
+            //Opens the catalog of packages that is available for the current user.
+            _packageCatalog = PackageCatalog.OpenForCurrentUser();
+        }
+
+        //
+        // Summary:
+        //     Indicates that an app package is installing.
+        public event TypedEventHandler<PackageCatalog, PackageInstallingEventArgs> PackageInstalling
+        {
+            add
+            {
+                _packageCatalog.PackageInstalling += value;
+            }
+
+            remove
+            {
+                _packageCatalog.PackageInstalling -= value;
+            }
+        }
+
+        //
+        // Summary:
+        //     Indicates that an app package is installing.
+        public event TypedEventHandler<PackageCatalog, PackageUninstallingEventArgs> PackageUninstalling
+        {
+            add
+            {
+                _packageCatalog.PackageUninstalling += value;
+            }
+
+            remove
+            {
+                _packageCatalog.PackageUninstalling -= value;
+            }
+        }
+
+
+        //
+        // Summary:
+        //     Indicates that an app package is installing.

```

Comment:
[nit] comment: installing -> updating. Also, there does not seem to be any `onPackageUpdating` event handler function. Do you think we should add it? Say on upgrading an app (and maybe the name changes) then would the updating event be triggered instead of the creating event?

Reply:
Fixed the nit.  It would be pretty easy to handle a the updating event.  We just need a use case / repo case to test against.   I'll try to find create this scenario when creating the functional tests, and fix it there.